full-heal.lua

```diff
@@ -162,9 +165,21 @@ if unit then
         status.severed_or_jammed = false
     end
 
-    for i, temp in ipairs(unit.status2.body_part_temperature) do
-        local bp = unit.body.body_plan.body_parts[i]
-        temp.whole = math.floor((bp.min_temp + bp.max_temp) / 2)
+    for i = #unit.status2.body_part_temperature-1,0,-1 do
+      unit.status2.body_part_temperature:erase(i) -- attempting to rewrite temperature was causing body parts to melt for some reason; forcing repopulation in this manner appears to be safer
```

Comment:
Any idea what this was setting the temperature to? Before I added it, body parts of creatures that had died by melting would *always* melt.

Reply:
Can't say I know. Your method sets body part temperatures to what are initially safe levels, but they proceed to rise abnormally in a gradual manner until melting point is reached; this appears to be especially likely for clothed (insulated) body parts.

Reply:
Ah, okay, it's possible I didn't notice that. If this also works for melted and resurrected creatures, it's fine with me.

Reply:
A few tests later, I can say that resurrecting creatures after having submerged them in magma appears to work without issue.